### PR TITLE
phpunit version update

### PR DESCRIPTION
**Pull request by hkulekci**
I update PHPUnit versiyon to ^5.5. But there is some issue on `UriTest.php` like this: 

```
PHPUnit 5.5.4 by Sebastian Bergmann and contributors.

...............................................................  63 / 807 (  7%)
............................................................... 126 / 807 ( 15%)
............................................................... 189 / 807 ( 23%)
............................................................... 252 / 807 ( 31%)
............................................................... 315 / 807 ( 39%)
............................................................... 378 / 807 ( 46%)
............................................................... 441 / 807 ( 54%)
............................................................... 504 / 807 ( 62%)
............................................................... 567 / 807 ( 70%)
............................................................... 630 / 807 ( 78%)
............................................................... 693 / 807 ( 85%)
............................................................... 756 / 807 ( 93%)
................................................FFF             807 / 807 (100%)

Time: 7.38 seconds, Memory: 21.75MB

There were 3 failures:

1) ZendTest\Diactoros\UriTest::testUtf8Uri
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'ουτοπία.δπθ.gr'
+'ο�_�_ο�_ία.δ�_θ.gr'

/Volumes/......../zend-diactoros/test/UriTest.php:569

2) ZendTest\Diactoros\UriTest::testUtf8Path with data set #0 ('http://example.com/тестовый_путь/', '/тестовый_путь/')
Failed asserting that null matches expected '/тестовый_путь/'.

/Volumes/......../zend-diactoros/test/UriTest.php:580

3) ZendTest\Diactoros\UriTest::testUtf8Path with data set #1 ('http://example.com/ουτοπία/', '/ουτοπία/')
Failed asserting that null matches expected '/ουτοπία/'.

/Volumes/......../zend-diactoros/test/UriTest.php:580

FAILURES!
Tests: 807, Assertions: 1367, Failures: 3.
```

I guess, they are not about PHPUnit version.  ping @weierophinney

### CODE REVIEW

**Comment by hkulekci** [comment]
I will update version number.

File: composer.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "psr/http-message": "~1.0"
   },
   "require-dev": {
-    "phpunit/PHPUnit": "~4.6",
+    "phpunit/phpunit": "^5.5",
```
Comment (weierophinney) [inline composer.json]: Change this constraint to `^4.6 || ^5.5` to allow this to test against PHP versions earlier than 5.5, please.

File: test/MessageTraitTest.php [hunk 2]
```diff
@@ -24,7 +24,7 @@ class MessageTraitTest extends TestCase
 
     public function setUp()
     {
-        $this->message = new Request(null, null, $this->getMock('Psr\Http\Message\StreamInterface'));
+        $this->message = new Request(null, null, $this->createMock('Psr\Http\Message\StreamInterface'));
```
Comment (weierophinney) [inline test/MessageTraitTest.php]: For this to work with both PHPUnit 4 and 5 versions, these will need to become:

```
$this->getMockBuilder(/* class or interface name */)->getMock()
```

**Comment by weierophinney** [comment]
Thanks, @hkulekci

**Pull request closed without merging**